services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/services/background_tasks_setup.py

```diff
@@ -0,0 +1,80 @@
+import asyncio
+import logging
+from collections.abc import Awaitable, Callable
+from typing import TypedDict
+
+from fastapi import FastAPI
+from servicelib.background_task import stop_periodic_task
+from servicelib.logging_utils import log_catch, log_context
+from servicelib.redis_utils import start_exclusive_periodic_task
+
+from ..core.settings import ApplicationSettings
+from .background_task_periodic_heartbeat_check import (
+    periodic_check_of_running_services_task,
+)
+from .modules.redis import get_redis_lock_client
+
+_logger = logging.getLogger(__name__)
+
+_TASK_NAME_PERIODICALY_CHECK_RUNNING_SERVICES = "periodic_check_of_running_services"
+
+
+class RutBackgroundTask(TypedDict):
+    name: str
+    task_func: Callable
+
+
+_RUT_BACKGROUND_TASKS = [
+    RutBackgroundTask(
+        name=_TASK_NAME_PERIODICALY_CHECK_RUNNING_SERVICES,
+        task_func=periodic_check_of_running_services_task,
+    )
+]
+
+
+def _on_app_startup(app: FastAPI) -> Callable[[], Awaitable[None]]:
+    async def _startup() -> None:
+        with log_context(
+            _logger, logging.INFO, msg="RUT background tasks startup.."
+        ), log_catch(_logger, reraise=False):
+            app_settings: ApplicationSettings = app.state.settings
+
+            app.state.rut_background_tasks = []
```

Comment:
Q: so why keep track of only 1 task in an array of tasks. What is the advantage of bouncing together all the tasks in one place instead of having each individual module mange its own?
I do not grasp the benefits of this design.